src/Shopping_Cart.html

```diff
@@ -32,19 +36,73 @@
+
+        Each row of this list, contains a button to remove items from the cart, that works in the following way:
+
+        2. Clicking on the "remove" button, updates an state object (`cartItem`), which, in turn, update two hidden fields on the form `form-cart-delete`, and also triggers a form submission. This form will call the server to update the cart, and update the `cartItemsList` object with the response.
+        2. The `amp-list` component contains the expression `[src]="cartItemsList.items"`, so that, when the `cartItemsList` object changes, as a result of the previous action, the list gets refreshed with the content of the updated cart.
+        -->
+        <amp-list credentials="include" layout="responsive" height="100px" width="500px" src="https://abe-cart-service.glitch.me/cart_items" [src]="cartItemsList.items">
+            <template type="amp-mustache" id="cart-items">
+                {{#isEmpty}}
+                <h3>Your Basket is Empty. </h3>
+                {{/isEmpty}}
+                {{^isEmpty}}
+                {{#cartItems}}
+                <div class="p1">{{name}} - {{price}}</div>
+                <div class="details pb1">
+                    <div class="px1">Color: {{color}}</div>
+                    <div class="px1">Size: {{size}}</div>
+                    <div class="px1">Qty: {{quantity}}</div>
+                    <button type="button" style="background: none;box-shadow: none;border-radius: 0px;width:30px" on="tap: AMP.setState({cartItem: 
+                                    { size: '{{size}}',
+                                      color: '{{color}}'
+                                    }}), form-cart-delete.submit">X</button>
+                </div>
+                <br />
+                {{/cartItems}}
+                {{/isEmpty}}
+            </template>
+        </amp-list>
     </div>
-    <p>Back to <a href="/samples_templates/product_page/preview/">product</a> sample.</p>
-  -->
+
+    <!-- ## Delete Items Form -->
+    <!-- 
+    As said, we use [amp-form](https://www.ampproject.org/docs/reference/components/amp-form) to send an `XHR POST` request to `/delete-cart-item`:
+
+    1. The form contains two hidden input fields, bound to the `cartItem` object's variables. When the remove button is clicked, this object gets updated with the details of the item to remove from cart, so the form can send them on the call to the server.
+    2. The form updates the `cartItemsList` with the response from the server, so that  `amp-list` can be refreshed with the contents of the updated cart.
+    -->
+    <form id="form-cart-delete" method="POST" target="_top" action-xhr="https://abe-cart-service.glitch.me/delete-cart-item" on="submit-success: AMP.setState({
+                cartItemsList: event.response
+            })" novalidate="">
+        <input type="hidden" name="color" value="" [value]="cartItem.color">
```

Comment:
I'm wondering whether we should add an id to the shopping cart items. Identifying items via color and size is not very intuitive.

Reply:
Good point.
The "product" state object doesn't contain Ids, but I've noticed the related [products response does](https://ampbyexample.com/json/related_products.json?__amp_source_origin=https%3A%2F%2Fampbyexample.com).
I could just add "1, 2, 3" to each product on the "product" object, and pass it on the form, so we can use it in the cart. 
Sounds good?

Reply:
perfect!

Reply:
Done! Also fixed the prices, as it was always sending the same value ($1.99).